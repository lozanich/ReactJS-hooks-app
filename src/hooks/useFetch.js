import { useEffect, useState, useRef } from "react";

export const useFetch = (url) => {
  const isMounted = useRef(true);
  const [data, setData] = useState({ data: [], loading: true, error: null });

  useEffect(() => {
    return () => {
      isMounted.current = false;
    };
  }, []);

  useEffect(() => {
    setData({ data: [], loading: true, error: null });

    fetch(url)
      .then((resp) => resp.json())
      .then((res) => {
        if (isMounted.current) {
          setData({
            data: res,
            loading: false,
            error: null,
          });
        }
      })
      .catch((error) => {
        setData({
          data: [],
          loading: false,
          error: error,
        });
      });
    // return () => {
    //   cleanup;
    // };
  }, [url]);

  return data;
};
