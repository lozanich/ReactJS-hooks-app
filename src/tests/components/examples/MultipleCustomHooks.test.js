import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { MultipleCustomHooks } from "./../../../components/examples/MultipleCustomHooks";
import { useFetch } from "../../../hooks/useFetch";
import { useCounter } from "../../../hooks/useCounter";
jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");

describe("Probando multiples hooks", () => {
  useCounter.mockReturnValue({
    counter: 1,
  });

  test("Debe de mostrarse correctamente ", () => {
    useFetch.mockReturnValue({ data: [], loading: true, error: null });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostar la informacion", () => {
    useFetch.mockReturnValue({
      data: [{ author: "Ivan", quote: "Hola mundo" }],
      loading: false,
      error: null,
    });

    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find(".mb-0").text().trim()).toBe("Hola mundo");
    expect(wrapper.find("footer").text().trim()).toBe("Ivan");
  });
});
