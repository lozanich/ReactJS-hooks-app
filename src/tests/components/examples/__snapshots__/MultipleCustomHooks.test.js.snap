// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando multiples hooks Debe de mostrarse correctamente  1`] = `
<Fragment>
  <h1>
    Breaking bad quotes!!!
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading....
  </div>
  <blockquote
    className="blockquote text-right"
  >
    <p
      className="mb-0"
    />
    <footer
      className="blockquote-footer"
    />
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Get more
  </button>
</Fragment>
`;
